src/browser/preload_scripts.ts

```diff
@@ -72,60 +74,46 @@ export function fetchAndLoadPreloadScripts(
     identity: Identity,
     preloadOption: PreloadOption,
     proceed?: () => void
-): Promise<undefined> {
-    let allLoaded: Promise<undefined>;
-
-    if (!preloadOption) {
-        preloadOption = [];
-    } else if (typeof preloadOption === 'string') {
-        // convert legacy `preloadOption` option into modern `preloadOption` option
-        preloadOption = [<PreloadInstance>{ url: preloadOption }];
-    }
-
-    if (!isPreloadOption(preloadOption)) {
-        const message = 'Expected `preload` option to be a string primitive OR an array of objects with `url` props.';
-        const err = new Error(message);
-        allLoaded = Promise.reject(err);
-    } else {
-        const loadedScripts: Promise<undefined>[] = preloadOption.map((preload: PreloadInstance) => {
-            updatePreloadState(identity, preload, 'load-started');
-
-            // following if clause avoids re-fetch for resources already in memory
-            // todo: following if clause slated for removal (RUN-3227, blocked by RUN-3162)
-            if (
-                !REGEX_FILE_SCHEME.test(preload.url) && // except when resource does NOT use "file" scheme...
-                System.getPreloadScript(preload.url)   // ...is resource already in memory?
-            ) {
-                // previously downloaded
-                updatePreloadState(identity, preload, 'load-succeeded');
-                return Promise.resolve();
-            } else {
-                // not previously downloaded *OR* previous downloaded failed
-                return fetch(identity, preload).then(load);
-            }
-        });
+): Promise<LoadResponses> {
+    let result: Promise<LoadResponses>;
+
+    const loadedScripts: Promise<undefined>[] = preloadOption.map((preload: PreloadInstance) => {
```

Comment:
@joneit this will be removed in the updated call right?

Reply:
yup, a lot of this is out of date. i'm merging in prior commits. it will change again in subsequent commits.